crates/brioche-core/src/script.rs

```diff
@@ -76,58 +132,9 @@ impl deno_core::ModuleLoader for BriocheModuleLoader {
         module_specifier: &deno_core::ModuleSpecifier,
         _maybe_referrer: Option<&deno_core::ModuleSpecifier>,
         _is_dyn_import: bool,
-    ) -> std::pin::Pin<Box<deno_core::ModuleSourceFuture>> {
-        let module_specifier: Result<BriocheModuleSpecifier, _> = module_specifier.try_into();
-        let sources = self.sources.clone();
-        let vfs = self.brioche.vfs.clone();
-        let future = async move {
-            let module_specifier = module_specifier?;
-            let contents = specifier::read_specifier_contents(&vfs, &module_specifier)?;
-
-            let code = std::str::from_utf8(&contents)
-                .context("failed to parse module contents as UTF-8 string")?;
-
-            let parsed = deno_ast::parse_module(deno_ast::ParseParams {
-                specifier: module_specifier.clone().into(),
-                text: code.into(),
-                media_type: deno_ast::MediaType::TypeScript,
-                capture_tokens: false,
-                scope_analysis: false,
-                maybe_syntax: None,
-            })?;
-            let transpiled = parsed.transpile(
-                &deno_ast::TranspileOptions {
-                    imports_not_used_as_values: deno_ast::ImportsNotUsedAsValues::Preserve,
-                    ..Default::default()
-                },
-                &deno_ast::EmitOptions {
-                    source_map: deno_ast::SourceMapOption::Separate,
-                    ..Default::default()
-                },
-            )?;
-
-            let mut sources = sources.borrow_mut();
-            if let Entry::Vacant(entry) = sources.entry(module_specifier.clone()) {
-                let source_map = transpiled
-                    .clone()
-                    .into_source()
-                    .source_map
-                    .context("source map not generated")?;
-                entry.insert(ModuleSource {
-                    source_contents: contents.clone(),
-                    source_map,
-                });
-            }
-
-            let module_specifier: deno_core::ModuleSpecifier = module_specifier.into();
-            Ok(deno_core::ModuleSource::new(
-                deno_core::ModuleType::JavaScript,
-                transpiled.into_source().into_string()?.text.into(),
-                &module_specifier,
-            ))
-        };
-
-        Box::pin(future)
+        _requested_module_type: deno_core::RequestedModuleType,
+    ) -> deno_core::ModuleLoadResponse {
+        deno_core::ModuleLoadResponse::Sync(self.load_module_source(module_specifier))
```

Comment:
Before that, we needed to pin a future here, since the function `load()` was only accepting an async result. We can now use a synchronous result. It let us remove this unnecessary async stuff.